lib/fpm/package/pacman.rb

```diff
@@ -0,0 +1,394 @@
+
+  # The list of supported compression types. Default is xz (LZMA2)
+  COMPRESSION_TYPES = [ "gz", "bzip2", "xz", "none" ]
+
+  option "--compression", "COMPRESSION", "The compression type to use, must " \
+    "be one of #{COMPRESSION_TYPES.join(", ")}.", :default => "xz" do |value|
+    if !COMPRESSION_TYPES.include?(value)
+      raise ArgumentError, "Pacman compression value of '#{value}' is invalid. " \
+        "Must be one of #{COMPRESSION_TYPES.join(", ")}"
+    end
+    value
+  end
+
+  def initialize(*args)
+    super(*args)
+    attributes[:pacman_opt_depends] = []
+  end # def initialize
+
+  def architecture
+    case @architecture
+      when nil
+        return %x{uname -m}.chomp   # default to current arch
+      when "amd64" # debian and pacman disagree on architecture names
+        return "x86_64"
+      when "native"
+        return %x{uname -m}.chomp   # 'native' is current arch
+      when "all", "any", "noarch"
+        return "any"
+      else
+        return @architecture
+    end
+  end # def architecture
+
+  def iteration
+    return @iteration ? @iteration : 1
+  end # def iteration
+
+  def config_files
+    return @config_files || []
+  end # def config_files
+
+  def dependencies
+    bogus_regex = /[^\sA-Za-z0-9><=-]/
+    # Actually modifies depencies if they are not right
+    bogus_dependencies = @dependencies.grep bogus_regex
+    if !bogus_dependencies.empty?
+      @dependencies.reject! { |a| a =~ bogus_regex }
+      logger.warn("Some of the dependencies looked like they weren't package " \
+                  "names. Such dependency entries only serve to confuse arch. " \
+                  "I am removing them.",
+                  :removed_dependencies => bogus_dependencies,
+                  :fixed_dependencies => @dependencies)
+    end
+    return @dependencies
+  end
+
+
+  # This method is invoked on a package when it has been converted to a new
+  # package format. The purpose of this method is to do any extra conversion
+  # steps, like translating dependency conditions, etc.
+  #def converted_from(origin)
+    # nothing to do by default. Subclasses may implement this.
+    # See the RPM package class for an example.
+  #end # def converted_from
+
+  # Add a new source to this package.
+  # The exact behavior depends on the kind of package being managed.
+  #
+  # For instance:
+  #
+  # * for FPM::Package::Dir, << expects a path to a directory or files.
+  # * for FPM::Package::RPM, << expects a path to an rpm.
+  #
+  # The idea is that you can keep pumping in new things to a package
+  # for later conversion or output.
+  #
+  # Implementations are expected to put files relevant to the 'input' in the
+  # staging_path
+  def input(pacman_pkg_path)
+    control = {}
+    # Unpack the control tarball
+    safesystem("tar", "-C", staging_path, "-xf", pacman_pkg_path)
+    pkginfo = File.join(staging_path, ".PKGINFO")
```

Comment:
Style recommendation: staging_path is a method that can take an argument. Mainly to avoid having to explicitly call File.join all over the place, you can do this:

```ruby
# Equivalent to File.join(staging_path, ".PKGINFO")
staging_path(".PKGINFO")
```